scripts/deploy/utils/git/tag/tag-utils.ts

```diff
@@ -3,12 +3,21 @@ import { githubRemote } from "./constants";
 
 /**
  * `createNewTag` is a helper function that creates a new tag.
+ * @returns true if tag creation is successful and false otherwise.
  */
 const createNewTag = () => {
-  execSync("yarn version --new-version patch", {
-    encoding: "utf-8",
-    stdio: "inherit",
-  });
+  console.log("Creating new tag...");
+  try {
+    execSync("yarn version --new-version patch", {
+      encoding: "utf-8",
+      stdio: "inherit",
+    });
+  } catch (err) {
+    console.log("Creating new tag failed.");
+    console.log("output", err);
+    return false;
+  }
+  return true;
```

Comment:
I think this composition is pretty confusing. Some util functions return these booleans indicating the status while others throw errors. What criteria is used to determine what should do what? Could we be a little more consistent in how we error? 
How do you feel about treating errors the way we do in go and wrapping them? 
https://stackoverflow.com/questions/42754270/re-throwing-exception-in-nodejs-and-not-losing-stack-trace